src/main/java/com/opentable/util/Optionals.java

```diff
@@ -74,6 +76,21 @@ private Optionals() {
         return first.isPresent() ? first : second.get();
     }
 
+    /**
+     * Combines two optionals to get a third one based on the values of the given ones if they are present.
+     * @param ox First optional.
+     * @param oy Second optional.
+     * @param combiner Function that combines the values of the given optionals.
+     * @param <X> type of first optional.
+     * @param <Y> type of the second optional.
+     * @param <Z> return type of the combiner function.
+     * @return The combined result wrapped into an optional if values of both optionals are present.
+     *         Optional.empty() otherwise.
+     */
+    public static <X, Y, Z> Optional<Z> combineWith(Optional<X> ox, Optional<Y> oy, BiFunction<X, Y, Z> combiner) {
```

Comment:
I'm using `X,Y,Z` since this is the pattern in all the file. Please confirm if this is ok, if not, to use `T1,T2..`
> You’ve done c but not a or b thans